data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java

```diff
@@ -107,15 +110,42 @@ PdlBuilder newBuilderInstance(Writer writer)
   private String _namespace = "";
   private String _package = "";
 
+  private final boolean _trackWriteLocations;
+
+  private final Map<Object, PdlSchemaParser.ParseLocation> _writeLocations;
+
   /**
    * Construct a .pdl source code encoder.
    *
    * @param out provides the encoded .pdl destination.
    */
   public SchemaToPdlEncoder(Writer out)
   {
-    _writer = out;
+    this(out, false);
+  }
+
+  /**
+   * Construct a .pdl source code encoder.
+   *
+   * @param out provides the encoded .pdl destination.
+   * @param returnContextLocations Enable recording the context locations of schema elements during parsing. The
+   *                              locations can be retrieved using {@link #getWriteLocations()} after parsing.
+   */
+  public SchemaToPdlEncoder(Writer out, boolean returnContextLocations)
+  {
+    if (returnContextLocations) {
+      _writeLocations = new IdentityHashMap<>();
+      // Wrap the Writer to track line/column numbers to report to elementWriteListener
+      LineColumnNumberWriter lineColumnNumberWriter = new LineColumnNumberWriter(out);
+      // When counting column numbers, CompactPDLBuilder treats ',' as whitespace
+      lineColumnNumberWriter.setIsWhitespaceFunction(c -> Character.isWhitespace(c) || c == ',');
```

Comment:
Should we set `|| c == ','` if encodingStyle is COMPACT?

Reply:
Good call. Updated.